src/Core/Stock/StockManager.php

```diff
@@ -389,8 +389,9 @@ private function prepareMovement($productId, $productAttributeId, $deltaQuantity
             $stockAvailable = $stockManager->getStockAvailableByProduct($product, $productAttributeId, $params['id_shop'] ?? null);
 
             if ($stockAvailable->id) {
+                $configuration = ServiceLocator::get('\\PrestaShop\\PrestaShop\\Core\\ConfigurationInterface');
```

Comment:
This is related to the block below `if (!empty($params['id_stock_mvt_reason'])) {` it should probably be executed only in an else statement